Assets/MRTK/Core/Providers/Hands/BaseHandVisualizer.cs

```diff
@@ -81,7 +81,10 @@ public bool TryGetJointTransform(TrackedHandJoint joint, out Transform jointTran
 
         void IMixedRealitySourceStateHandler.OnSourceLost(SourceStateEventData eventData)
         {
-            if (Controller?.InputSource.SourceId == eventData.SourceId)
+            // We must check if either this or gameObject equate to null because this callback may be triggered after
+            // the object has been destroyed. Although event handlers are unregistered in OnDisable(), this may in fact
+            // be postponed (see BaseEventSystem.UnregisterHandler()).
+            if ((MonoBehaviour)this != null && gameObject != null && Controller?.InputSource.SourceId == eventData.SourceId)Controller?.InputSource.SourceId == eventData.SourceId)
```

Comment:
why is the cast to MonoBehaviour necessary?

Reply:
Also wondering this throughout some of these scripts. `this` is already a `BaseHandVisualizer`, which is itself a `MonoBehaviour`. The cast seems unnecessary (and VS agrees) 
![image](https://user-images.githubusercontent.com/3580640/141217441-bcc80191-9913-4dd8-9c91-2968a015782a.png)

![image](https://user-images.githubusercontent.com/3580640/141217482-b3e408cf-c736-46f7-ac16-296937679949.png)

Reply:
This is to workaround a quirk of how Unity overloads the `==` operator: depending on whether or not the method is called though an interface or a concrete object, the `this == null` check may not in fact test for object deletion. Casting to `MonoBehaviour` forces the correct overload to be used.

Reply:
Have a look at:
 - https://answers.unity.com/questions/586144/destroyed-monobehaviour-not-comparing-to-null.html
 - https://stackoverflow.com/questions/49852414/dont-use-interfaces-in-unity

Reply:
That part I know, but this _is_ a `MonoBehaviour` directly, not an interface

Reply:
We have a helper extension for checking `null` on interfaces that might be `MonoBehaviour`s (this matches a suggestion in your first link): https://github.com/microsoft/MixedRealityToolkit-Unity/blob/9000dfed1d2bdfdb0e5d88b319c49a3f8700037c/Assets/MRTK/Core/Extensions/UnityObjectExtensions.cs#L53-L58

Reply:
Oh, I wasn't aware of that, I'll change the checks to use that instead